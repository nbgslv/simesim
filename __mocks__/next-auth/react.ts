const mockSession = {
  expires: new Date(Date.now() + 2 * 86400).toISOString(),
  user: {
    email: '0512345678',
    emailEmail: 'test@test.com',
    id: 'clcclcclcclcclcclc1234',
    image: null,
    name: 'foo bar',
    role: 'ADMIN',
  },
};

export const useSession = jest.fn(
  () => ({ data: mockSession, status: 'authenticated' }) // return type is [] in v3 but changed to {} in v4
);
export const signOut = jest.fn(() =>
  Promise.resolve({ url: process.env.NEXTAUTH_URL as string })
);
