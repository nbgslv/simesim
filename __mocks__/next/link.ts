import React from 'react';

const Link = ({
  children,
  href,
}: {
  children: React.ReactNode;
  href: string;
}) =>
  React.cloneElement(
    children as React.ReactElement,
    { href, 'data-testid': href } as any
  );

export default Link;
