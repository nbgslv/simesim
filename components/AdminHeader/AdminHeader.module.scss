@import 'styles/variables';

.main {
  :global {
    .navbar-nav a {
      color: $text-color;

      &:hover {
        color: $primary-color;
      }
    }
  }
}
