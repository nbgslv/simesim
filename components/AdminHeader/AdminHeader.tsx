import React from 'react';
import { Nav, Navbar } from 'react-bootstrap';
import Image from 'next/image';
import { signOut } from 'next-auth/react';
import styles from './AdminHeader.module.scss';

const AdminHeader = () => (
  <Navbar className={styles.main}>
    <Navbar.Toggle aria-controls="basic-navbar-nav" />
    <Navbar.Collapse id="basic-navbar-nav">
      <Nav>
        <Nav.Link eventKey="/admin/main" href="/admin/main">
          ראשי
        </Nav.Link>
        <Nav.Link eventKey="/admin/plans" href="/admin/plans">
          תכניות
        </Nav.Link>
        <Nav.Link eventKey="/admin/lines" href="/admin/lines">
          קווים
        </Nav.Link>
        <Nav.Link eventKey="/admin/bundles" href="/admin/bundles">
          חבילות
        </Nav.Link>
        <Nav.Link eventKey="/admin/plansModel" href="/admin/plansModel">
          מודלים
        </Nav.Link>
        <Nav.Link eventKey="/admin/countries" href="/admin/countries">
          מדינות
        </Nav.Link>
        <Nav.Link eventKey="/admin/coupons" href="/admin/coupons">
          קופונים
        </Nav.Link>
        <Nav.Link eventKey="/admin/users" href="/admin/users">
          משתמשים
        </Nav.Link>
        <Nav.Link eventKey="/admin/inquiries" href="/admin/inquiries">
          פניות
        </Nav.Link>
        <Nav.Link eventKey="/admin/apikeys" href="/admin/apikeys">
          API
        </Nav.Link>
        <Nav.Link disabled eventKey="/settings" href="/admin/main">
          הגדרות
        </Nav.Link>
      </Nav>
      <Nav className="me-auto">
        <Nav.Link onClick={() => signOut()}>התנתק</Nav.Link>
        <Nav.Link href="/">חזרה לאתר</Nav.Link>
      </Nav>
    </Navbar.Collapse>
    <Navbar.Brand>
      <Image src="/logo.png" alt="logo" width={80} height={50} />
    </Navbar.Brand>
  </Navbar>
);

export default AdminHeader;
