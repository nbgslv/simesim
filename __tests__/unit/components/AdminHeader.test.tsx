import { render, screen, fireEvent } from '@testing-library/react';
import { signOut } from 'next-auth/react';
import * as Sentry from '@sentry/nextjs';
import { Router } from 'next/router';
import React from 'react';
import AdminHeader from '../../../components/AdminHeader/AdminHeader';

describe('AdminHeader', () => {
  const spies: any = {};

  beforeEach(() => {
    jest.clearAllMocks();
    spies.routerChangeStart = jest.fn();
    Router.events.on('routeChangeStart', spies.routerChangeStart);
  });

  afterEach(() => {
    Router.events.off('routeChangeStart', spies.routerChangeStart);
  });

  it('matches snapshot', () => {
    const { container } = render(<AdminHeader />);
    expect(container).toMatchSnapshot();
  });

  it('renders the header with all navigation links', () => {
    render(<AdminHeader />);

    expect(screen.getAllByRole('link')).toHaveLength(15);
  });

  it('clicking the sign out link calls Sentry.setUser and signOut', () => {
    const setUserMock = jest.spyOn(Sentry, 'setUser');

    render(<AdminHeader />);
    fireEvent.click(screen.getByText('התנתק'));

    expect(setUserMock).toHaveBeenCalledTimes(1);
    expect(setUserMock).toHaveBeenCalledWith(null);
    expect(signOut).toHaveBeenCalledTimes(1);
  });

  // TODO next router navigation(link clicks)
});
