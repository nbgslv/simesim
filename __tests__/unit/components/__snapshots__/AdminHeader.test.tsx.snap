// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHeader matches snapshot 1`] = `
<div>
  <nav
    class="main navbar navbar-expand navbar-light"
  >
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/main"
          data-testid="/admin/main"
          href="/admin/main"
        >
          ראשי
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/plans"
          data-testid="/admin/plans"
          href="/admin/plans"
        >
          תכניות
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/lines"
          data-testid="/admin/lines"
          href="/admin/lines"
        >
          קווים
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/bundles"
          data-testid="/admin/bundles"
          href="/admin/bundles"
        >
          חבילות
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/plansModel"
          data-testid="/admin/plansModel"
          href="/admin/plansModel"
        >
          מודלים
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/countries"
          data-testid="/admin/countries"
          href="/admin/countries"
        >
          מדינות
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/coupons"
          data-testid="/admin/coupons"
          href="/admin/coupons"
        >
          קופונים
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/transactions"
          data-testid="/admin/transactions"
          href="/admin/transactions"
        >
          תנועות
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/users"
          data-testid="/admin/users"
          href="/admin/users"
        >
          משתמשים
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/supportedphones"
          data-testid="/admin/supportedphones"
          href="/admin/supportedphones"
        >
          טלפונים נתמכים
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/blog"
          data-testid="/admin/blog"
          href="/admin/blog"
        >
          בלוג
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/inquiries"
          data-testid="/admin/inquiries"
          href="/admin/inquiries"
        >
          פניות
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/admin/apikeys"
          data-testid="/admin/apikeys"
          href="/admin/apikeys"
        >
          API
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/settings"
          data-testid="/admin/settings"
          href="/admin/settings"
        >
          הגדרות
        </a>
      </div>
      <div
        class="me-auto navbar-nav"
      >
        <a
          class="nav-link"
          href="#"
          role="button"
          tabindex="0"
        >
          התנתק
        </a>
        <a
          class="nav-link"
          data-rr-ui-event-key="/"
          data-testid="/"
          href="/"
        >
          חזרה לאתר
        </a>
      </div>
    </div>
    <span
      class="navbar-brand"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2750%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="logo"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </span>
  </nav>
</div>
`;
